pkg/clustermanager/reconcile_logicalcluster.go

```diff
@@ -45,17 +45,60 @@ func (c *controller) reconcileLogicalCluster(key string) error {
 		}
 	} else {
 		c.logger.V(2).Info("reconcile LogicalCluster", "resource", cluster.Name)
-		err := c.processAddLC(cluster)
+		err := c.processAddOrUpdateLC(cluster)
 		if err != nil {
 			return err
 		}
 	}
 	return nil
 }
 
-// TODO: the processAddLC needs to check whether the LogicalCluster has already
-// been added and processed, or whether this is actually an Update.
-func (c *controller) processAddLC(newCluster *lcv1alpha1.LogicalCluster) error {
+// processAddOrUpdateLC: process an added LC object
+func (c *controller) processAddOrUpdateLC(logicalCluster *lcv1alpha1.LogicalCluster) error {
+	logger := klog.FromContext(c.context)
+	var err error
+
+	switch logicalCluster.Status.Phase {
+	case lcv1alpha1.LogicalClusterPhaseEmpty:
+		// Either the client created a new logical cluster object and expects us to
+		// manage/create a physical cluster, or the discovery identified a new
+		// physical cluster.
+		return c.processNewLC(logicalCluster)
+	case lcv1alpha1.LogicalClusterPhaseInitializing:
+		// A managed cluster is being created by the provider. Need to wait for
+		// the cluster to be created at which point discovery will change the
+		// state to READY and update the cluster config.
+		return nil
+	case lcv1alpha1.LogicalClusterPhaseNotReady:
+		// Discovery noticed that an cluster has disappeared
+		// TODO: The current logic is missing a finalizer for managed logical cluster
+		if logicalCluster.Spec.Managed {
+			err := errors.New("a managed physical cluster has been removed")
+			return err
+		}
+		return c.clientset.
+			LogicalclusterV1alpha1().
+			LogicalClusters(GetNamespace(logicalCluster.Spec.ClusterProviderDescName)).
+			Delete(c.context, logicalCluster.Name, v1.DeleteOptions{})
+	case lcv1alpha1.LogicalClusterPhaseReady:
+		// The cluster is ready - we have nothing to do but celebrate!
+		return nil
+	default:
+		err = errors.New("unknown logical cluster phase")
+		logger.Error(err, string(logicalCluster.Status.Phase))
+		return err
+	}
```

Comment:
Handle the initial empty phase here. There is a return in switch case for every other status.

Reply:
done